jadx-gui/src/main/java/jadx/gui/ui/codearea/FridaAction.java

```diff
@@ -32,157 +32,134 @@
-	private String methodName;
+	private final Map<String, Boolean> isInitial = new HashMap<>();
 
 	public FridaAction(CodeArea codeArea) {
 
 		super(NLS.str("popup.frida") + " (f)", codeArea);
-		LOG.info("triggered meee");
 		KeyStroke key = getKeyStroke(KeyEvent.VK_F, 0);
 		codeArea.getInputMap().put(key, "trigger frida");
 		codeArea.getActionMap().put("trigger frida", new AbstractAction() {
 			@Override
 
 			public void actionPerformed(ActionEvent e) {
 				node = getNodeByOffset(codeArea.getWordStart(codeArea.getCaretPosition()));
-				copyFridaCode();
+
+				String fridaSnippet = generateFridaSnippet();
+
+				Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+				StringSelection selection = new StringSelection(fridaSnippet);
+				clipboard.setContents(selection, selection);
 			}
 		});
 	}
 
-	private void copyFridaCode() {
-
-		if (node != null) {
-			if (node instanceof JMethod) {
-				JMethod n = (JMethod) node;
-				MethodNode methodNode = n.getJavaMethod().getMethodNode();
-				MethodInfo mi = methodNode.getMethodInfo();
-				methodName = mi.getName();
-				if (methodName.equals("<init>") || methodName.equals("onCreate")) {
-					methodName = "$init";
-				}
-				String fullClassName = methodNode.getParentClass().getFullName();
-				String className = methodNode.getParentClass().getShortName();
-				LOG.debug("node is jmethod");
-				ClassNode tmp = methodNode.getParentClass();
-				while (true) {
-					if (!tmp.isTopClass()) {
-						fullClassName = fullClassName.substring(0, fullClassName.lastIndexOf(".")) + "$"
-								+ fullClassName.substring(fullClassName.lastIndexOf(".") + 1, fullClassName.length());
-					} else {
-						break;
-					}
-					tmp = tmp.getParentClass();
-				}
-				JMethod jMth = (JMethod) node;
-				int mthLine = jMth.getLine();
-				List<String> argNames = jMth.getRootClass().getCodeInfo().getAnnotations().entrySet().stream()
-						.filter(e -> e.getKey().getLine() == mthLine && e.getValue() instanceof VarDeclareRef)
+	private String generateFridaSnippet() {
+		if (node instanceof JMethod) {
+			LOG.debug("node is jmethod");
+			return generateMethodSnippet((JMethod) node);
+		} else if (node instanceof JClass) {
+			LOG.debug("node is jclass");
+			return generateClassSnippet((JClass) node);
+		} else if (node instanceof JField) {
+			LOG.debug("node is jfield");
+			return generateFieldSnippet((JField) node);
+		}
+		LOG.debug("cannot generate frida snippet from node");
+		return "";
+
+	}
+
+	private String generateMethodSnippet(JMethod jMth) {
+		JavaMethod javaMethod = jMth.getJavaMethod();
+		MethodInfo methodInfo = javaMethod.getMethodNode().getMethodInfo();
+		String methodName = methodInfo.getName();
+		if (methodName.equals("<init>") || methodName.equals("onCreate")) {
+			methodName = "$init";
+		}
+		String rawClassName = javaMethod.getDeclaringClass().getRawName();
+		String shortClassName = javaMethod.getDeclaringClass().getName();
```

Comment:
```suggestion
		String shortClassName = javaMethod.getDeclaringClass().getClassNode().getClassInfo().getShortName();
```
Is here should be raw short name?
If yes, maybe it is better to add method `getShortRawName()` to `JavaClass` class.

Reply:
I think it's better to keep `javaMethod.getDeclaringClass().getName()` because if the class has an alias it returns it instead of the short name.
This way we can use any renaming the user did for the variable name that holds the class, while still referring to the actual class in the arguments to the Frida functions.
For example if I renamed a class, and then generate a snippet I would get something like this: `let renameClass = Java.use("com.example.rawClass")`

Reply:
Ok, that's fine